wc-cosigner/MultiKeySigner.sol

```diff
@@ -0,0 +1,149 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.23;
+
+import { ECDSA } from "solady/utils/ECDSA.sol";
+import "./passkey.sol";
+import "contracts/interfaces/ISigner.sol";
+import { SubLib } from "contracts/lib/SubLib.sol";
+import "forge-std/console2.sol";
+
+struct Config {
+    address eoaSigner;
+    WebAuthnValidatorData passkeySigner;
+}
+
+enum SignerType {
+    EOA,
+    PASSKEY
+}
+
+struct Signer {
+    SignerType signerType;
+    bytes data;
+}
+
+library SignerEncode {
+    function decodeEOA(Signer memory signer) internal pure returns (address eoa) {
+        eoa = address(bytes20(signer.data));
+    }
+
+    function decodePasskey(Signer memory signer) internal pure returns (WebAuthnValidatorData memory passkey) {
+        passkey = abi.decode(signer.data, (WebAuthnValidatorData));
+    }
+
+    function encodeSigners(Signer[] memory signers) internal pure returns (bytes memory encoded) {
```

Comment:
we should maybe expose this function as a public function in the contract (not here in the library) 

might help the frontend guys to encode it properly